### PR TITLE
Product Page

### PR DESCRIPTION
**Pull request by LuongHuy**

### CODE REVIEW

File: online-grocery-store/src/components/cart/CartList.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const CartList = (props) => {
       <Table.Td>
         <QuantityButton
           quantity={item.quantity}
-          addProduct={() => addProduct(item)}
+          addProduct={(() => addProduct(item), 1)}
```
Comment (dustinle48) [inline online-grocery-store/src/components/cart/CartList.js]: `addProduct={() => addProduct(item,1)}`

File: online-grocery-store/src/pages/product/[product_id].js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+} from "@mantine/core";
+import { useCart } from "../../store/store";
+import { Rating } from "@mantine/core";
+import Link from "next/link";
+
+const ProductPage = () => {
+  const addProduct = useCart((state) => state.addProduct);
+  const cart = useCart((state) => state.cart);
+  const [quantity, setQuantity] = useState(0);
+  console.log(cart);
```
Comment (dustinle48) [inline online-grocery-store/src/pages/product/[product_id].js]: remove log

File: online-grocery-store/src/pages/product/[product_id].js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        setProduct(product);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [router.query]);
+  const decreaseQuantity = () => {
+    if (quantity > 0) {
+      setQuantity(quantity - 1);
```
Comment (dustinle48) [inline online-grocery-store/src/pages/product/[product_id].js]: Read [this](https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates)
`setQuantity(prevQuantity => prevQuantity > 0 ? prevQuantity - 1 : 0)`

File: online-grocery-store/src/pages/product/[product_id].js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  };
+
+  const increaseQuantity = () => {
+    setQuantity(quantity + 1);
```
Comment (dustinle48) [inline online-grocery-store/src/pages/product/[product_id].js]: `setQuantity(prevQuantity => prevQuantity + 1)`

File: online-grocery-store/src/pages/shoppingcart.js [hunk 5]
```diff
@@ -12,6 +13,12 @@ const ShoppingCart = () => {
 
   return (
     <>
+      <Head>
+        <title>Create Next App</title>
+        <meta name="description" content="Generated by create next app" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <link rel="icon" href="Logo.jpg" />
```
Comment (dustinle48) [inline online-grocery-store/src/pages/shoppingcart.js]: `src` not `href`

**Review by dustinle48 — CHANGES_REQUESTED** [review]

File: online-grocery-store/src/components/cart/CartList.js (hunk 1 above)
Comment (LuongHuy) [reply online-grocery-store/src/components/cart/CartList.js]: Done

File: online-grocery-store/src/pages/product/[product_id].js (hunk 2 above)
Comment (LuongHuy) [reply online-grocery-store/src/pages/product/[product_id].js]: Done

File: online-grocery-store/src/pages/product/[product_id].js (hunk 3 above)
Comment (LuongHuy) [reply online-grocery-store/src/pages/product/[product_id].js]: Done

File: online-grocery-store/src/pages/product/[product_id].js (hunk 4 above)
Comment (LuongHuy) [reply online-grocery-store/src/pages/product/[product_id].js]: Done

File: online-grocery-store/src/pages/shoppingcart.js (hunk 5 above)
Comment (LuongHuy) [reply online-grocery-store/src/pages/shoppingcart.js]: Done

**Pull request merged**